src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/commands.py

```diff
@@ -181,6 +183,7 @@ def load_command_table(self, _):
         g.custom_command('list-consumption-locations', 'list_consumption_locations')
         g.custom_command('identity assign', 'assign_identity')
         g.custom_command('identity show', 'show_identity')
+        g.generic_update_command('update', getter_name='get_app', setter_name='update_functionapp', command_type=appservice_custom)
```

Comment:
To preserve the generic update "contract", you need to separate the setter (the PUT) from processing the convenience arguments. Otherwise, the convenience arguments will trump the generic arguments in the event of a conflict, which is opposite of other updates.

So you'll need to specify the `custom_func_name` here to expose the convenience arguments.